presto-main/src/main/java/com/facebook/presto/sql/ParsingUtil.java

```diff
@@ -14,29 +14,42 @@
 package com.facebook.presto.sql;
 
 import com.facebook.presto.Session;
+import com.facebook.presto.execution.warnings.WarningCollector;
+import com.facebook.presto.spi.PrestoWarning;
 import com.facebook.presto.sql.parser.ParsingOptions;
-import com.facebook.presto.sql.parser.ParsingWarning;
-
-import java.util.function.Consumer;
+import static com.facebook.presto.sql.parser.ParsingOptions.DecimalLiteralTreatment;
 
 import static com.facebook.presto.SystemSessionProperties.isParseDecimalLiteralsAsDouble;
+import static com.facebook.presto.spi.StandardWarningCode.PARSER_WARNING;
 import static com.facebook.presto.sql.parser.ParsingOptions.DecimalLiteralTreatment.AS_DECIMAL;
 import static com.facebook.presto.sql.parser.ParsingOptions.DecimalLiteralTreatment.AS_DOUBLE;
 
 public class ParsingUtil
 {
+    private ParsingUtil() {}
+
     public static ParsingOptions createParsingOptions(Session session)
     {
-        return new ParsingOptions(isParseDecimalLiteralsAsDouble(session) ? AS_DOUBLE : AS_DECIMAL);
+        return ParsingOptions.builder()
```

Comment:
You can delegate to `public static ParsingOptions createParsingOptions(Session session, WarningCollector warningCollector)` suppying NOOP warning collector